src/components/Pill/PillButtonEnhanced/PillButtonEnhanced.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import DownArrow from '@material-design-icons/svg/round/expand_more.svg';
+import UpArrow from '@material-design-icons/svg/round/expand_less.svg';
+import Clear from '@material-design-icons/svg/round/cancel.svg';
+import { bem } from '../../../utils';
+import { ENTER_KEY } from '../../../constants';
+import { PillButtonBaseProps } from '../PillButton';
+import { PrioritySelector, PrioritySelectorProps } from '../../PrioritySelector';
+import styles from './PillButtonEnhanced.scss';
+import { Tooltip } from '../../Tooltip';
+
+export interface Props<PriorityItemValue = unknown> extends PillButtonBaseProps {
+    /** Label/indicator in case of multiple selection, such as `+2` for instance */
+    additionalContentLabel?: string;
+    /** Tooltip content to be shown when multiple selections are shown */
+    additionalContentTooltip?: React.ReactNode;
+    /** Props needed for rendering optional priority selector */
+    priority?: PrioritySelectorProps<PriorityItemValue>;
+    /** max width of the button (excluding 20px of the trailing button), e.g. `fit-content`. If undefined will default to 220px */
+    maxWidth?: string;
+}
+
+const { block, elem } = bem('PillButtonEnhanced', styles);
+
+export const PillButtonEnhanced = React.forwardRef(
+    /* eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint */
+    <PriorityItemValue extends unknown>(
+        {
+            isOpen = false,
+            toggleDropdown,
+            onClear,
+            name,
+            content = null,
+            downArrowLabel,
+            upArrowLabel,
+            clearLabel,
+            priority,
+            additionalContentLabel,
+            additionalContentTooltip,
+            maxWidth,
+            ...rest
+        }: Props<PriorityItemValue>,
+        ref: React.Ref<HTMLElement>
+    ) => {
+        const isActive = !!content;
+        const propsForBem = { isOpen, isActive };
+
+        let buttonIcon = (
+            <DownArrow aria-label={downArrowLabel} viewBox="0 0 24 24" height="14px" width="14px" />
+        );
+        let handleButtonClick = (e) => {
+            e.stopPropagation();
+            e.preventDefault();
+            toggleDropdown();
+        };
+        if (isOpen) {
+            buttonIcon = (
+                <UpArrow aria-label={upArrowLabel} viewBox="0 0 24 24" height="14px" width="14px" />
+            );
+        }
+        if (content) {
+            buttonIcon = (
+                <Clear aria-label={clearLabel} viewBox="0 0 24 24" height="14px" width="14px" />
+            );
+            handleButtonClick = (e) => {
+                e.stopPropagation();
+                e.preventDefault();
+                onClear();
+            };
+        }
```

Comment:
it's not a necessary change, just a suggestion 
It is a little bit confusing, maybe it is just me 😃 Maybe some kind of helper function which would be reposinsible for the button would make it cleaner:
Something in this direction:
```
  const getButtonIconAndHandler = () => {
            if (content) {
                return {
                    icon: (
                        <Clear
                            aria-label={clearLabel}
                            viewBox="0 0 24 24"
                            height="14px"
                            width="14px"
                        />
                    ),
                    onClick: (e: React.MouseEvent) => {
                        e.stopPropagation();
                        e.preventDefault();
                        onClear();
                    },
                };
            }
            if (isOpen) {
                return {
                    icon: (
                        <UpArrow
                            aria-label={upArrowLabel}
                            viewBox="0 0 24 24"
                            height="14px"
                            width="14px"
                        />
                    ),
                    onClick: (e: React.MouseEvent) => {
                        e.stopPropagation();
                        e.preventDefault();
                        toggleDropdown();
                    },
                };
            }
            return {
                icon: (
                    <DownArrow
                        aria-label={downArrowLabel}
                        viewBox="0 0 24 24"
                        height="14px"
                        width="14px"
                    />
                ),
                onClick: (e) => {
                    e.stopPropagation();
                    e.preventDefault();
                    toggleDropdown();
                },
            };
        };

        const { icon: buttonIcon, onClick: handleButtonClick } = getButtonIconAndHandler();
```
You can even further simplify it. But it is up to you, just a suggestion :-)